### PR TITLE
TST: Unblock Appveyor build by patching `subprocess`

### LINKED ISSUE #9176: Appveyor build failing
The appveyor build is failing fairly consistently on the python `3.5` build (eg. see https://ci.appveyor.com/project/matplotlib/matplotlib/build/1.0.2957/job/dqhano9lykge7ffb#L1375). The actual failure is somewhere in `test_artist`:

```
lib\matplotlib\tests\test_artist.py ...ssCommand exited with code -1073740791
```

### PR DESCRIPTION
**Pull request by Kojoley**
closes #9176

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -95,6 +95,12 @@ install:
   # pytest-cov>=2.3.1 due to https://github.com/pytest-dev/pytest-cov/issues/124
   - pip install -q pytest "pytest-cov>=2.3.1" pytest-rerunfailures pytest-timeout
 
+  # Apply patch to `subprocess` on Python versions > 2 and < 3.6.3
+  # https://github.com/matplotlib/matplotlib/issues/9176
+  - python -c "import sys; sys.exit((2,) > sys.version_info > (3,6,3))" && (
```
Comment (tacaswell) [inline .appveyor.yml]: Shouldn't this be `sys.exit((3,6,3) > sys.version_info > (2,))` ?

Comment (Kojoley) [reply .appveyor.yml]: Yes, I wrongly negated the `(2,) < sys.version_info < (3,6,3)` expression.

File: .appveyor.yml [hunk 2]
```diff
@@ -95,6 +95,12 @@ install:
   # pytest-cov>=2.3.1 due to https://github.com/pytest-dev/pytest-cov/issues/124
   - pip install -q pytest "pytest-cov>=2.3.1" pytest-rerunfailures pytest-timeout
 
+  # Apply patch to `subprocess` on Python versions > 2 and < 3.6.3
+  # https://github.com/matplotlib/matplotlib/issues/9176
+  - python -c "import sys; sys.exit(not (2,) < sys.version_info < (3,6,3))" && (
+    curl -sL https://github.com/python/cpython/pull/1224.patch |
+    patch -fsup 1 -d %CONDA_PREFIX% ) || ( set errorlevel= )
```
Comment (anntzer) [inline .appveyor.yml]: I thought only Py3 was affected?  Both from my (possibly incorrect) remembering of what used to fail and from https://bugs.python.org/issue30121#msg300658
If it's only a py3 issue then the lower bound should be (3,); if not we can just drop the lower bound (it's not as if we will ever support py1.5 anyways :-)).

Comment (Kojoley) [reply .appveyor.yml]: The bug was fixed 4 years ago in Py2. Python >= 2.7.6 should not be affected.

**Pull request merged**